doc/source/dev/contributor/devpy_test.rst

```diff
@@ -0,0 +1,145 @@
+:orphan:
+
+.. _devpy-runtest:
+
+===========================
+Running SciPy Tests Locally
+===========================
+
+Basic test writing and execution from within the Python interpreter is
+documented in the `NumPy/SciPy Testing Guidelines`_. This page includes
+information about running tests from the command line using SciPy's
+``dev.py`` command line tool. *Note: Before beginning, ensure that* |pytest|_
+*is installed.*
+
+To run all tests, navigate to the root SciPy directory at the command
+line and execute
+
```

Comment:
I suggest adding a `.. note::` here that the `dev.py` interface is self-documenting, so everything on this page and more (including usage examples for each command) can be accessed with `python dev.py --help` and for individual commands like `python dev.py <command-name> --help`.